packages/cli/src/lib/utils/string.ts

```diff
@@ -0,0 +1,3 @@
+export function camelToSnakeCase(str: string) {
+  return str.replace(/[A-Z]/g, (letter) => `_${letter.toLowerCase()}`);
+}
```

Comment:
https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABBAhgWwKYBsAqcDKYKA1hgMIoDOGAFJVAE4Bci9DMYA5gJSIDeAKESIGGKCAZI2AOlEAHLCgi0A9AG0AggFoAWgF0VnADSIaWMVAwNeAXgB8iAAYB9ACR9zUSw2lQ4AGTgAdysKahpuAF9HbgBuAUiBAQgESjhzaSw4ThpUTFwCIlIw2gByAAUUKCqMUu5uIA

Just to prevent user-derp, I'd do:

```suggestion
export function camelToSnakeCase(str: string) {
  if(/^[A-Z]/.test(str)) {
    throw `'${str}' is not a camelCase string.`;
  }
  return str.replace(/[A-Z]/g, (letter) => `_${letter.toLowerCase()}`);
}
```

Reply:
Instead of throwing an error, why not removing leading and trailing underscores.
This function is used before sending analytic events. So throwing errors is not a good idea.